### PR TITLE
feat: wait for Executor shutdown

### LINKED ISSUE #2372: Wait for Executor shutdown
Query tasks are run on background threadpools, however, these are never actually shutdown. Whilst likely harmless, this should be fixed.

### PR DESCRIPTION
**Pull request by crepererum**
Closes #2372.

### CODE REVIEW

File: query/src/exec.rs [hunk 1]
```diff
@@ -176,7 +197,8 @@ mod tests {
         let expected_strings = to_set(&["Foo", "Bar"]);
         let plan = StringSetPlan::Known(Arc::clone(&expected_strings));
 
-        let ctx = Executor::new(1).new_context(ExecutorType::Query);
+        let exec = Executor::new(1);
```
Comment (alamb) [inline query/src/exec.rs:200]: this change (to name `exec`) is to ensure that `exec` lives to the end of this block, right?

File: src/influxdb_ioxd.rs [hunk 2]
```diff
@@ -341,6 +341,9 @@ async fn serve(
 
     info!("server completed shutting down");
 
+    application.join();
+    info!("application completed shutting down");
```
Comment (alamb) [inline src/influxdb_ioxd.rs:345]:
```suggestion
    info!("shared application state completed shutting down");
```

**Review by alamb — APPROVED** [review]
Looks great -- thank you @crepererum

File: query/src/exec.rs (hunk 1 above)
Comment (crepererum) [reply query/src/exec.rs:200]: yes.

**Pull request merged**